### DISCUSSION TITLE
console.log printing in terminal console not browser console

### DISCUSSION THREAD

**Discussion by shadowbrand** [vercel/next.js#44917] [Help] [answered]
On this repo: 
https://github.com/sanity-io/nextjs-blog-cms-sanity-v3

If I insert `console.log('test')` on line 19 of components/IndexPage.tsx I would assume this would print in the browser, but it prints in the vscode's console only.

Is there something wrong with my environment, or did I miss something? Please, enlighten me.

**Comment by icyJoseph** [comment, marked as answer]
When using the Server Components API, that is, appDir, components are rendered on the Server, where JSX is transformed to an HTML string. 

If there's no interactivity required for a page, or component sub-tree, no runtime is shipped, and the component is not even hydrated client side, so any logging done won't show up on the client. The bare minimum needed by the browser to show content is an HTML string, and that's what Server Components aim to ship, by default. Of course CSS and other resources might be part of the payload, but not a runtime, by default. If needed, one has to opt-in view the client directive.

So the answer is that all of the rendering is done server side, and no code is executed on the client, unless you provide the `use client` directive.
